packages/tx/src/baseTransaction.ts

```diff
@@ -237,18 +260,10 @@ export abstract class BaseTransaction<TransactionObject> {
 
   public isSigned(): boolean {
     const { v, r, s } = this
-    if (this.type === 0) {
-      if (!v || !r || !s) {
-        return false
-      } else {
-        return true
-      }
+    if (v === undefined || r === undefined || s === undefined) {
+      return false
     } else {
-      if (v === undefined || !r || !s) {
-        return false
-      } else {
-        return true
-      }
+      return true
```

Comment:
Is this really totally safe to say that this code can be simplified like this with v, r, s now being represented as bigint? This code part is really critical.

(again, would be good to have a look on occasion if tests are covering all cases and in doubt another test case would also not hurt)

Reply:
yes this should be better code by checking undefined.

i remember we had a bug that led to this code in the first place so i had to look it up 😄 https://github.com/ethereumjs/ethereumjs-monorepo/commit/3f8f382e1c1cff74ca61077b7ad9e56fa148272a where EIP2930 supports v's with value 0 (empty Buffer)

it looks like we have a good amount of `isSigned()` tests in `test/legacy.spec.ts:462`, but only for legacy tx. this would be a good candidate to move to `base.spec.ts` and iterate through txTypes array. i'll try to do soon.